server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/lightningbitcoin/LightningBitcoinExtension.java

```diff
@@ -43,34 +44,48 @@ public String getName() {
 
     @Override
     public IWallet createWallet(String walletLogin, String tunnelPassword) {
-        if (walletLogin != null && !walletLogin.trim().isEmpty()) {
-            StringTokenizer st = new StringTokenizer(walletLogin, ":");
-            String walletType = st.nextToken();
+        try {
+            if (walletLogin != null && !walletLogin.trim().isEmpty()) {
+                StringTokenizer st = new StringTokenizer(walletLogin, ":");
+                String walletType = st.nextToken();
 
-            if ("eclair".equalsIgnoreCase(walletType)) {
-                String scheme = st.nextToken();
-                String host = st.nextToken();
-                String port = st.nextToken();
-                String password = st.nextToken();
-                if (scheme != null && host != null && port != null && password != null) {
-                    return new EclairWallet(scheme, host, Integer.parseInt(port), password);
-                }
-            } else if ("lnd".equalsIgnoreCase(walletType)) {
-                // echo 127.0.0.1:8080:`xxd -ps -u -c10000 ~/.lnd/data/chain/bitcoin/mainnet/admin.macaroon`:`xxd -ps -u -c10000 ~/.lnd/tls.cert`
-                String host = st.nextToken();
-                String port = st.nextToken();
-                String macaroon = st.nextToken();
-                String cert = st.nextToken();
-                if (host != null && port != null && macaroon != null && macaroon.length() > 4 && cert != null && cert.length() > 4) {
-                    try {
-                        return new LndWallet(host, Integer.parseInt(port), macaroon, cert);
-                    } catch (Exception e) {
-                        log.warn("Error creating lnd wallet; host={}, port={}, macaroon={}..., cert={}...", host, port, macaroon.substring(0, 4), cert.substring(0, 4), e);
+                if ("eclair".equalsIgnoreCase(walletType)) {
+                    String scheme = st.nextToken();
+                    String host = st.nextToken();
+                    int port = Integer.parseInt(st.nextToken());
+                    String password = st.nextToken();
+
+                    InetSocketAddress tunnelAddress = ctx.getTunnelManager().connectIfNeeded(tunnelPassword, InetSocketAddress.createUnresolved(host, port));
+                    host = tunnelAddress.getHostString();
+                    port = tunnelAddress.getPort();
```

Comment:
the diff doesn't show it properly, these 3 lines are whats new, its indented because of the added try/catch